src/main/java/mcp/mobius/waila/addons/thermalexpansion/HUDHandlerEnergyCell.java

```diff
@@ -15,6 +16,13 @@
 
 public class HUDHandlerEnergyCell implements IWailaDataProvider {
 
+
+    private final DecimalFormat energyFormat;
+
+    public HUDHandlerEnergyCell() {
+        this.energyFormat = new DecimalFormat("###,###,###,###,###,###,###");
+    }
```

Comment:
I don't think this needs to be an instance variable instead of a static, and should use NumberFormat. The below sets the (locale-specific from system) group separators and removes fractional digits (since I think RF is entirely integer).
```suggestion
    private static final NumberFormat energyFormat = NumberFormat.instance()
    
    static {
        energyFormat.setGroupingUsed(true);
        energyFormat.setMaximumFractionDigits(0);
    }
```